### DISCUSSION TITLE
add toggle method to menu

### DISCUSSION THREAD

**Discussion by patrickwasp** [zauberzeug/nicegui#860] [Ideas / Feature Requests]
quassar has a method for toggling menus. It would be nice to also have it supported here.

so something like this 
```python
with ui.button(on_click=lambda: menu.close() if menu.value else menu.open()).props('icon=menu'):
    with ui.menu().props('persistent auto-close') as menu:
        ui.label('content')
```

could look like this
```python
with ui.button(on_click=lambda: menu.toggle()).props('icon=menu'):
    with ui.menu().props('persistent auto-close') as menu:
        ui.label('content')

```

https://quasar.dev/vue-components/menu

**Comment by rodja** [comment]
I am a big fan of "toggle". Let us implement it everywhere :-)
Beside `ui.menu` there are other element in NiceGUI which would benefit from a toggle function: `ui.drawer`, `ui.switch`, `ui.checkbox`, ...
Maybe a `Toggleable` [mixin](https://github.com/zauberzeug/nicegui/tree/main/nicegui/elements/mixins) could implement this in a generic way? Anyone interested in contributing? Pull requests are welcome.

**Reply by falkoschindler** [reply]
Drawers already have a toggle method.

**Comment by falkoschindler** [comment]
I just added `ui.menu.toggle()`.
Of course this shouldn't prevent us from introducing this method for other element or even a mixin if needed. For now I'll close this feature request as completed.
